### PR TITLE
Add setter for colorbar label text

### PR DESCRIPTION
**Pull request by djhoese**
Let's try this again. This is a replacement for #2056 which after trying to rebase and merge just got too ugly with commit history. I instead copied the current state of #1292 into a new branch and committed with @macrocosme as the author of the commit.

Updates .label property so it can be provided a TextVisual or a new string to update the existing TextVisual of a ColorbarVisual.

This is a replacement for #1292 so that I can actually make commits to it (compared to the original fork's master branch).

I'll need to reapply the style fixes from `main`.

### CODE REVIEW

**Comment by djhoese** [comment]
@kmuehlbauer This should be ready to go if you have time to review it.

**Comment by kmuehlbauer** [comment]
> @kmuehlbauer This should be ready to go if you have time to review it.

I'm already afk, but will have a look tomorrow morning.

File: vispy/visuals/colorbar.py [hunk 1]
```diff
@@ -255,7 +253,7 @@ class ColorBarVisual(CompoundVisual):
     pos : tuple (x, y)
         Position where the colorbar is to be placed with
         respect to the center of the colorbar
-    label_str : str
+    label : str
```
Comment (kmuehlbauer) [inline vispy/visuals/colorbar.py:256]: I'm wondering if we should allow here `TextVisual` too, as we do below in the `label.setter`?

**Review by kmuehlbauer — APPROVED** [review]
I have just one remark. Otherwise LGTM.

File: vispy/scene/widgets/colorbar.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def __init__(self, cmap, orientation,
 
         self._colorbar = ColorBarVisual(size=dummy_size, cmap=cmap,
                                         orientation=orientation,
-                                        label_str=label, clim=clim,
+                                        label_text=label, clim=clim,
```
Comment (almarklein) [inline vispy/scene/widgets/colorbar.py:81]: This should be just `label`, right?

Comment (djhoese) [reply vispy/scene/widgets/colorbar.py:81]: Yeah...looks like we need another test.

**Comment by djhoese** [comment]
@kmuehlbauer @almarklein Ok the last set of commits do a few things:

1. Fix the colorbar widget so it passes label properly
2. Add tests for the colorbar widget (there weren't any)
3. Add the ability to pass a TextVisual to the ColorbarVisual for the label (and add a test)
4. Remove `**kwargs` from the ColorbarVisual. I'm not really sure why this is here (maybe some of the tests will end up failing), but I'd confidently say we don't want `**kwargs` unless it is really really necessary.

**Comment by djhoese** [comment]
Ok, tests pass now. This should be ready for re-review and merge.

File: vispy/visuals/colorbar.py [hunk 3]
```diff
@@ -280,28 +284,27 @@ class ColorBarVisual(CompoundVisual):
...
         self._cmap = get_colormap(cmap)
         self._clim = clim
         self._pos = pos
         self._size = size
         self._orientation = orientation
 
-        self._label = TextVisual(self._label_str, color=self._label_color)
+        if not isinstance(label, TextVisual):
+            label = TextVisual(label, color=label_color)
+        self._label = label
```
Comment (kmuehlbauer) [inline vispy/visuals/colorbar.py:301]: Do we need to override a given TextVisual's color with the given `label_color`? So if the user gives a `TextVisual` which is in `green` and also gives `label_color="red"`. Or should we just make clear in the docstring, that `label_color` is only used when `label` is a string?

**Review by kmuehlbauer — COMMENTED** [review]
Just one more remark on the `label_color`.

File: vispy/visuals/colorbar.py (hunk 3 above)
Comment (almarklein) [reply vispy/visuals/colorbar.py:301]: > Or should we just make clear in the docstring, that label_color is only used when label is a string?

That's what happens now, right? :)

Comment (almarklein) [reply vispy/visuals/colorbar.py:301]: or do you mean the ticks below? Maybe these should use `label.color` instead?

File: vispy/visuals/colorbar.py [hunk 4]
```diff
@@ -255,9 +253,10 @@ class ColorBarVisual(CompoundVisual):
...
         Position where the colorbar is to be placed with
         respect to the center of the colorbar
-    label_str : str
+    label : str | vispy.visuals.TextVisual
         The label that is to be drawn with the colorbar
         that provides information about the colorbar.
+        If a TextVisual object then 'label_color' is ignored.
     label_color : str | vispy.color.Color
         The color of the labels. This can either be a
         str as the color's name or an actual instace of a vipy.color.Color
```
Comment (almarklein) [inline vispy/visuals/colorbar.py:262]: This also determines the color of the ticks.

File: vispy/visuals/colorbar.py (hunk 3 above)
Comment (almarklein) [reply vispy/visuals/colorbar.py:301]: If so, we should also update the tick colors when the label setter is given a label object.

**Review by almarklein — CHANGES_REQUESTED** [review]
Only comment is to mention ticks in docs for `label_color`.

File: vispy/visuals/colorbar.py (hunk 3 above)
Comment (kmuehlbauer) [reply vispy/visuals/colorbar.py:301]: So many options... We might need to add setters for all that stuff...

Comment (djhoese) [reply vispy/visuals/colorbar.py:301]: I added a note in the `label` docstring that color is not used if it is a TextVisual. I think every Visual in VisPy needs to make their properties more available. Not sure this PR is going to be the place though. Might be something good for a hacktoberfest issue.

File: vispy/visuals/colorbar.py (hunk 4 above)
Comment (djhoese) [reply vispy/visuals/colorbar.py:262]: Ooohhh. :thinking: My added statement isn't exactly wrong though as far as the color of the label goes.

File: vispy/visuals/colorbar.py [hunk 5]
```diff
@@ -255,9 +253,10 @@ class ColorBarVisual(CompoundVisual):
     pos : tuple (x, y)
         Position where the colorbar is to be placed with
         respect to the center of the colorbar
-    label_str : str
+    label : str | vispy.visuals.TextVisual
         The label that is to be drawn with the colorbar
         that provides information about the colorbar.
+        If a TextVisual object then 'label_color' is ignored.
     label_color : str | vispy.color.Color
         The color of the labels. This can either be a
```
Comment (djhoese) [inline vispy/visuals/colorbar.py:261]:
```suggestion
        The color of the label and tick labels. This can either be a
```

**Review by almarklein — APPROVED** [review]

**Pull request merged**